### ISSUE TITLE
[WinRM::FS::Core::FileTransporter] Upload failed

### ISSUE THREAD

**Issue by milanoid** [mitchellh/vagrant#7717]
### Vagrant version

Vagrant 1.8.5
### Host operating system

Windows 10 with VirtualBox 5.0.26 r108824
### Guest operating system

Windows Server 2012 R2
### Vagrantfile

``` ruby
Vagrant.configure("2") do |config|
  config.vm.box = "my-box"
  config.vm.provision "file", source: "my_script.cmd", destination: "my_script.cmd"
end
```
### Debug output

https://gist.github.com/milanoid/a341572b96f8e9a9b5821817cfefe30d
### Expected behavior

File copied over to guest.
### Actual behavior

Upload error.
### Steps to reproduce
1. `vagrant reload --provision --debug`

**Comment by git-jiby-me** [comment]
@chrisroberts the problem here is the powershell scrips in winrm-fs, they use types like `System.Security.Cryptography.MD5CryptoServiceProvider` and `System.Security.Cryptography.FromBase64Transform` which are not available in windows nano. I was able to manually tweak these scripts in my machine, to use alternatives and provisioning was working. 

Can we update `winrm` and `winrm-fs` to the latest versions from upstream, as there are changes in how it handles file uploads there, and then i could test with the latest `winrm-fs` version and submit patches for the powershell scripts if necessary

**Comment by chrisroberts** [comment]
Related: #8102

**Comment by chrisroberts** [comment]
Fixed via #8102
